File: openapi/openapi.json/code_samples/typescript/sessions/post.ts
import { AsId, CreateSession, PortsSpec, QosProfile, TestSdk, UeId } from 'test-sdk';

(async () => {
  const testSdk = new TestSdk({
    token: 'YOUR_TOKEN',
  });

  const ueId: UeId = {
    externalId: '123456789@domain.com',
    msisdn: '123456789',
    ipv4addr: '192.168.0.1/24',
    ipv6addr: '2001:db8:85a3:8d3:1319:8a2e:370:7344',
  };

  const asId: AsId = {
    ipv4addr: '192.168.0.1/24',
    ipv6addr: '2001:db8:85a3:8d3:1319:8a2e:370:7344',
  };

  const ranges: Ranges = {
    from: 15507,
    to: 5265,
  };

  const portsSpec: PortsSpec = {
    ranges: [ranges],
    ports: [7941],
  };

  const qosProfile = QosProfile.QOSE;

  const createSession: CreateSession = {
    duration: 86400,
    ueId: ueId,
    asId: asId,
    uePorts: portsSpec,
    asPorts: portsSpec,
    qos: qosProfile,
    notificationUrl: 'https://application-server.com',
    notificationAuthToken: 'c8974e592c2fa383d4a3960714',
  };

  const { data } = await testSdk.qoSSessions.createSession(input, {
    authorization: 'Authorization',
    xAuthorization: 'X-Authorization',
    xTransactionId: 'X-transaction-id',
  });

  console.log(data);
})();

File: openapi/code_samples/python/sessions/post.py
from test_sdk import TestSdk, Environment
from test_sdk.models import CreateSession

sdk = TestSdk(
    access_token="YOUR_ACCESS_TOKEN",
    base_url=Environment.DEFAULT.value,
    timeout=10000
)

request_body = CreateSession(
    duration=86400,
    ue_id={
        "external_id": "123456789@domain.com",
        "msisdn": "123456789",
        "ipv4addr": "192.168.0.1/24",
        "ipv6addr": "2001:db8:85a3:8d3:1319:8a2e:370:7344"
    },
    as_id={
        "ipv4addr": "192.168.0.1/24",
        "ipv6addr": "2001:db8:85a3:8d3:1319:8a2e:370:7344"
    },
    ue_ports={
        "ranges": [
            {
                "from_": 15507,
                "to": 5265
            }
        ],
        "ports": [
            7941
        ]
    },
    as_ports={
        "ranges": [
            {
                "from_": 15507,
                "to": 5265
            }
        ],
        "ports": [
            7941
        ]
    },
    qos="QOS_E",
    notification_url="https://application-server.com",
    notification_auth_token="c8974e592c2fa383d4a3960714"
)

result = sdk.qo_s_sessions.create_session(
    request_body=request_body,
    authorization="Authorization",
    x_authorization="X-Authorization",
    x_transaction_id="X-transaction-id"
)

print(result)